### ISSUE TITLE
Sometimes when the menu is opened an item is selected before moving the mouse

### ISSUE THREAD

**Issue by bradydean** [Schneegans/Fly-Pie#206]
About 25% of the time when I open the menu, an item is selected. If I let go of my modifier key then the item is opened. This is pretty annoying. If I wiggle my mouse a bit then the item is unselected then menu behaves as normal - an item isn't selected again until my mouse moves out of the inner radius.

I don't think this is a config thing. My inner radius is set such that my mouse must leave the center circle before an item should be selected.

1. Open menu (ctrl+space)
2. See if item is selected

## Expected Behavior
An item is not selected immediately after opening the menu.

## System
 - Linux distribution: Fedora 35
 - Fly-Pie version: 13
 - GNOME Shell version: 41.3

**Comment by Schneegans** [comment]
Hi there! Thanks for the report! However, this never happened to me, so this may get difficult to debug. Two questions come to my mind:
* Are you using Wayland or X11? Maybe you can try switching this and see if it makes a difference. Input handling is quite different on both systems.
* Do you have any other input device attached? Maybe a touchpad, a touchscreen, or a tablet? If so, you could try disabling / unplugging this and see if it makes any difference...

**Comment by bradydean** [comment]
Same issue on Xorg. Maybe that cuts out all of the tricky input handling code from debugging :)

My mouse is the only input device.

**Comment by bradydean** [comment]
Here's my settings

```
{
  "easing-duration": 0.25,
  "easing-mode": "ease-out",
  "background-color": "rgba(0, 0, 0, 0.13)",
  "text-color": "rgb(238, 238, 238)",
  "font": "Cantarell 11",
  "wedge-width": 664,
  "wedge-inner-radius": 79,
  "wedge-color": "rgba(0, 0, 0, 0.13)",
  "wedge-color-hover": "rgba(255, 255, 255, 0.014)",
  "wedge-separator-color": "rgba(38, 38, 38, 0.32)",
  "wedge-separator-width": 1,
  "center-color-mode": "fixed",
  "center-color-mode-hover": "fixed",
  "center-fixed-color": "rgba(255, 255, 255, 0)",
  "center-fixed-color-hover": "rgba(255, 255, 255, 0)",
  "center-auto-color-saturation": 0.75,
  "center-auto-color-saturation-hover": 0.75,
  "center-auto-color-luminance": 0.8,
  "center-auto-color-luminance-hover": 0.8,
  "center-auto-color-opacity": 0,
  "center-auto-color-opacity-hover": 0,
  "center-size": 168,
  "center-size-hover": 168,
  "center-icon-scale": 0.55,
  "center-icon-scale-hover": 0.78,
  "center-icon-opacity": 0,
  "center-icon-opacity-hover": 1,
  "center-icon-crop": 0.93,
  "center-icon-crop-hover": 0.75,
  "center-background-image": "assets/default.svg",
  "center-background-image-hover": "assets/default.svg",
  "child-color-mode": "fixed",
  "child-color-mode-hover": "auto",
  "child-fixed-color": "rgba(255, 255, 255, 0)",
  "child-fixed-color-hover": "rgba(255, 255, 255, 0)",
  "child-auto-color-saturation": 0.26,
  "child-auto-color-saturation-hover": 0.9,
  "child-auto-color-luminance": 0.17,
  "child-auto-color-luminance-hover": 0.2,
  "child-auto-color-opacity": 1,
  "child-auto-color-opacity-hover": 1,
  "child-size": 84,
  "child-size-hover": 84,
  "child-offset": 143,
  "child-offset-hover": 143,
  "child-icon-scale": 0.73,
  "child-icon-scale-hover": 0.73,
  "child-icon-opacity": 1,
  "child-icon-opacity-hover": 1,
  "child-icon-crop": 0.8,
  "child-icon-crop-hover": 0.8,
  "child-background-image": "assets/default-child.svg",
  "child-background-image-hover": "assets/default-child-highlight.svg",
  "child-draw-above": false,
  "grandchild-color-mode": "fixed",
  "grandchild-color-mode-hover": "fixed",
  "grandchild-fixed-color": "rgba(255, 255, 255, 0)",
  "grandchild-fixed-color-hover": "rgba(255, 255, 255, 0)",
  "grandchild-size": 20,
  "grandchild-size-hover": 23,
  "grandchild-offset": 24,
  "grandchild-offset-hover": 28,
  "grandchild-background-image": "assets/default-grandchild.svg",
  "grandchild-background-image-hover": "assets/default-grandchild.svg",
  "grandchild-draw-above": false,
  "trace-min-length": 200,
  "trace-thickness": 8,
  "trace-color": "rgb(38, 38, 38)",
  "touch-buttons-opacity": 0.3
}
```

I use the favorites menu and have 10 items.

**Comment by bradydean** [comment]
https://github.com/Schneegans/Fly-Pie/blob/c717d5f0cd425349f0ca5897265503814cff11d1/src/extension/SelectionWedges.js#L365-L379

`ok` is never checked before calculating `distance`. Perhaps the transformation failed and causes the conditional to be taken?

**Comment by Schneegans** [comment]
That's really weird. So just to verify that I understood it correctly:
* You move your mouse to any position on the screen.
* Then you keep the mouse stationary for a short time (just to make things easier)
* Then you press <kbd>Ctrl</kbd>+<kbd>Space</kbd>
* The menu opens centered at the mouse pointer.
* A random (?) item may look as it is hovered by the mouse.
* When you move the mouse a little, the item becomes un-hovered.

Thankfully, you seem to have the knowledge required to debug this on your end :stuck_out_tongue_winking_eye:. I think we will have to add some debug prints to the code in order to see what's going on... If you can confirm that I understood everything correctly, I will give you some hints where you could put some debug prints to the code...

**Comment by Schneegans** [comment]
I will attach an external monitor later today and see if I can reproduce this. I think you are already quite close to the solution here!

**Comment by Schneegans** [comment]
I was able to reproduce this and found the bug. The `onMotionEvent` of the `SelectionWedges` expects global coordinates but the `_setPosition` of the `Menu` provided monitor-local coordinates. So it only worked properly on the left-most monitor.

I also fixed the annoying transition which smoothly moved the menu position from the last monitor to the current one.

Feel free do test the `develop` branch! Hopefully, it is working well on your end.

Thanks again for the report!
